clerk.go

```diff
@@ -169,6 +169,11 @@ type BackendConfig struct {
 	// Key is the Clerk secret key. If it's not set, the package level
 	// secretKey will be used.
 	Key *string
+	// Application allows you to provide information for your application
+	// that will be set in API request headers. Good candidates are the
+	// application's domain or ID, but possible values are not limited to
+	// those.
+	Application *string
```

Comment:
❓ I think it might be best to nest this option under a `DefaultHeaders` struct?

Reply:
You mean something like this, or a `map[string][]string`?

```go
type BackendConfig struct {
  DefaultHeaders Headers 
}

type Headers struct {
  Application string
}
```

I guess you're thinking that we might want to be able to add more custom headers in the future right?

Reply:
I was thinking the same exact structure as in your snippet (and avoid a `map`). Struct fields can be more informative/self-documented in a way. 
I think there's already a project that may introduce a default request header (not sure if we plan to provide it through this struct though but could be an option).

Reply:
Updated. Please take another look, thanks!